pyat/at/lattice/utils.py

```diff
@@ -734,3 +741,62 @@ def set_shift(ring: Sequence[Element], dxs, dzs, relative=False) -> None:
     dzs = numpy.broadcast_to(dzs, (len(ring),))
     for el, dx, dy in zip(ring, dxs, dzs):
         shift_elem(el, dx, dy, relative=relative)
+
+
+def get_geometry(ring: Sequence[Element], start_coordinates=(0, 0, 0),
+                 centered=False, offset=(0.0, 0.0)):
+    """Compute the 2D ring geometry in cartesian coordinates
+
+    Parameters:
+        ring: Lattice description
+        start_coordinates: x,y,angle at starting point
+        centered: it True the coordinates origin is the center of the ring
+        offset: (dx, dy) offsets coordinates by the given amount
+
+    Returns:
+        geomdata: recarray containing, x, y, angle
+        radius: machine radius
+    """
+    def isDipole(el):
+        if isinstance(el, Dipole):
+            if el.BendingAngle != 0.0:
+                return True
+        return False
+
+    geom_dtype = [('x', numpy.float64, (1, )),
+                  ('y', numpy.float64, (1, )),
+                  ('angle', numpy.float64, (1, ))]
+    geomdata = numpy.recarray((len(ring)+1, ), dtype=geom_dtype)
+    xx = numpy.zeros((len(ring)+1, 1))
+    yy = numpy.zeros((len(ring)+1, 1))
+    angle = numpy.zeros((len(ring)+1, 1))
+    x, y, t = start_coordinates
+
+    for ind, el in enumerate(ring+[ring[0]]):
+        ang = 0.0
+        ll = el.Length
+        if isDipole(el):
+            ang = 0.5 * el.BendingAngle
+            ll *= numpy.sin(ang)/ang
+        t -= ang
+        x += ll * numpy.cos(t)
+        y += ll * numpy.sin(t)
+        t -= ang
+        xx[ind] = x
+        yy[ind] = y
+        angle[ind] = t
+
+    try:
+        radius = ll / abs(t)
+    except ValueError as ex:
+        print(ex)
+        print('geometry: radius not computed, set to 0.0')
```

Comment:
Is this printout necessary? If so, it should be printed only if `centered` is True. Otherwise it doesn't matter. And should preferably converted in a warning, or printed to stderr.